core/src/Cuda/Kokkos_Cuda_Instance.cpp

```diff
@@ -428,11 +428,13 @@ void *CudaInternal::resize_team_scratch_space(int scratch_pool_id,
   // Multiple ParallelFor/Reduce Teams can call this function at the same time
   // and invalidate the m_team_scratch_ptr. We use a pool to avoid any race
   // condition.
+  auto mem_space = Kokkos::CudaSpace::impl_create(m_cudaDev, m_stream);
+  Kokkos::Cuda exec_space(m_stream);
```

Comment:
Is this odd to create a Kokkos::Cuda object inside `CudaInternal`? We could always implement this function in `Cuda` class since we have access to the execution space where this is called, and just hand the calling Cuda object to `kokkos_malloc/realloc` below.

Reply:
I would prefer if we just used `mem_space.allocate` and `mem_space.deallocate` here as we do for the other scratch allocations. This doesn't require adding an overload for `kokkos_malloc` with an execution space which we could decide on elsewhere.

Reply:
and yes, I don't think it's a good idea to create a `Cuda` execution space instance from within `CudaInternal`.